staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -230,6 +242,10 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	for {
 		select {
+		case <-s.ServerShuttingdownCh:
+			// the server has signaled that it is shutting down (not accepting
+			// any new request), all active watch request will return now.
```

Comment:
nit:

... will return over the grace period

or sth like that?

Reply:
updated the comment